### PR TITLE
Jobs templates

### PR DESCRIPTION
**Pull request by jwierzbo**

### CODE REVIEW

File: pkg/apis/ml/v1alpha1/extension_spec_job.go [hunk 1]
```diff
@@ -25,31 +25,60 @@ import (
...
+}
+
+func (j *ArangoMLJobsTemplates) Validate() error {
+	if j == nil {
+		return nil
+	}
+
+	var errs []error
+	for _, template := range j.Prediction {
+		if err := template.Validate(); err != nil {
```
Comment (nikita-vanyasin) [inline pkg/apis/ml/v1alpha1/extension_spec_job.go:43]: No need to check for errors - `shared.WithErrors` will combine only non-nil entries anyway

**Review by nikita-vanyasin — APPROVED** [review]

File: pkg/apis/ml/v1alpha1/extension_spec_job.go [hunk 2]
```diff
@@ -25,31 +25,56 @@ import (
 	sharedApi "github.com/arangodb/kube-arangodb/pkg/apis/shared/v1"
 )
 
-type ArangoMLExtensionSpecInit struct {
+type ArangoMLJobsTemplates struct {
+	// Prediction defines template for the prediction job
+	Prediction map[string]*ArangoMLExtensionTemplateSpec `json:"prediction,omitempty"`
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/extension_spec_job.go:30]: Do not use map, use CPU/GPU fields instead

**Review by ajanikow — CHANGES_REQUESTED** [review]

File: pkg/apis/ml/v1alpha1/extension_conditions.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ package v1alpha1
 import api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1"
 
 const (
+	MLExtensionFoundCondition              api.ConditionType = "MLExtensionFound"
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/extension_conditions.go:26]: To ExtensionFoundCondition + move to conditions.go

File: pkg/apis/ml/v1alpha1/extension_spec_job.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+		return nil
+	}
+
+	return shared.WithErrors(
+		shared.PrefixResourceErrors("cpu", a.CPU.Validate()),
+		shared.PrefixResourceErrors("gpu", a.GPU.Validate()),
+	)
+}
+
+type ArangoMLExtensionTemplateSpec struct {
```
Comment (ajanikow) [inline pkg/apis/ml/v1alpha1/extension_spec_job.go:106]: ArangoMLExtensionTemplateSpec -> Może ArangoMLExtensionPodTemplate ?

**Review by ajanikow — APPROVED** [review]

**Pull request merged**